canonical-web-and-design/maas-ui
# feat(ui): add compose and refresh actions to LXD project action bar

## Done

- Added compose and refresh actions to LXD project action bar
- Added types for KVM actions

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the project tab of a LXD KVM
- Click on "Compose VM" in the VM table action bar and check that the compose VM form opens
- Click on the refresh button in the VM table action bar and check that the refresh KVM form opens
- With any form open, change to a different KVM details tab and check that the form closes

## Fixes

Fixes #2235
